File: ios/Sources/MultandoSDK/Models/Enums.swift
import Foundation

/// Status of a traffic violation report.
public enum ReportStatus: String, Codable, Sendable {
    case pending
    case underReview = "under_review"
    case verified
    case rejected
    case appealed
    case resolved
    /// The community (via votes / reputation-weighted validation) has
    /// flagged the report as credible, but no authority has acted on it.
    case communityVerified = "community_verified"
    /// The report has been forwarded to an authority for formal review.
    case authorityReview = "authority_review"
}

/// Source that originated a report.
public enum ReportSource: String, Codable, Sendable {
    case sdk
    case web
    case mobile
    case api
}

/// High-level vehicle category.
public enum VehicleCategory: String, Codable, Sendable {
    case car
    case motorcycle
    case truck
    case bus
    case bicycle
    case scooter
    case other
}

/// Type of evidence attached to a report.
public enum EvidenceType: String, Codable, Sendable {
    case photo
    case video
    case audio
    case document
}

File: android/multando-sdk/src/main/kotlin/com/multando/sdk/models/Enums.kt
package com.multando.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class ReportStatus {
    @SerialName("pending") PENDING,
    @SerialName("under_review") UNDER_REVIEW,
    @SerialName("community_verified") COMMUNITY_VERIFIED,
    @SerialName("authority_review") AUTHORITY_REVIEW,
    @SerialName("verified") VERIFIED,
    @SerialName("rejected") REJECTED,
    @SerialName("appealed") APPEALED,
    @SerialName("resolved") RESOLVED
}

@Serializable
enum class ReportSource {
    @SerialName("sdk") SDK,
    @SerialName("web") WEB,
    @SerialName("mobile") MOBILE,
    @SerialName("api") API
}

@Serializable
enum class VehicleCategory {
    @SerialName("car") CAR,
    @SerialName("motorcycle") MOTORCYCLE,
    @SerialName("truck") TRUCK,
    @SerialName("bus") BUS,
    @SerialName("bicycle") BICYCLE,
    @SerialName("scooter") SCOOTER,
    @SerialName("other") OTHER
}

@Serializable
enum class EvidenceType {
    @SerialName("photo") PHOTO,
    @SerialName("video") VIDEO,
    @SerialName("audio") AUDIO,
    @SerialName("document") DOCUMENT
}
